io/ringbuffer_test.go

```diff
@@ -318,3 +318,74 @@ func TestRingBuffer_ExhaustiveRead(t *testing.T) {
 		t.Errorf("Expected ring buffer size to be %v, got %v", e, a)
 	}
 }
+
+func TestRingBuffer_Reset(t *testing.T) {
+	byteSlice := make([]byte, 10)
+	ringBuffer := NewRingBuffer(byteSlice)
+
+	ringBuffer.Write([]byte("Hello-world"))
+	if ringBuffer.size == 0 {
+		t.Errorf("expected ringBuffer to not be empty")
+	}
+
+	readBuffer := make([]byte, 5)
+	ringBuffer.Read(readBuffer)
+	if ringBuffer.size == 0 {
+		t.Errorf("expected ringBuffer to not be empty")
+	}
+	if e, a := "ello-", string(readBuffer); !strings.EqualFold(e, a) {
+		t.Errorf("expected read string to be %s, got %s", e, a)
+	}
+
+	// reset the buffer
+	ringBuffer.Reset()
+	if e, a := 0, ringBuffer.size; e != a {
+		t.Errorf("expect default size to be %v , got %v", e, a)
+	}
+	if e, a := 0, ringBuffer.start; e != a {
+		t.Errorf("expect deafult start to point to %v , got %v", e, a)
+	}
+	if e, a := 0, ringBuffer.end; e != a {
+		t.Errorf("expect default end to point to %v , got %v", e, a)
+	}
+	if e, a := 10, len(ringBuffer.slice); e!=a {
+		t.Errorf("expect ringBuffer capacity to be %v, got %v", e, a)
+	}
+
+	ringBuffer.Write([]byte("someThing new"))
+	if ringBuffer.size == 0 {
+		t.Errorf("expected ringBuffer to not be empty")
+	}
+
+	ringBuffer.Read(readBuffer)
+	if ringBuffer.size == 0 {
+		t.Errorf("expected ringBuffer to not be empty")
+	}
+	if e, a := "eThin", string(readBuffer); !strings.EqualFold(e, a) {
+		t.Errorf("expected read string to be %s, got %s", e, a)
+	}
```

Comment:
Would be helpful to add comment here why `eThin` is the expected result instead something like `someT`

Reply:
Updated